### PR TITLE
Cache programming classes 

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Follows the same pattern as in #46118 to cache ProgrammingClass pages. These pages should be static once they hit production, so there's no reason to hit the database each time. I also added caching for programming_environments#get_summary_by_name while I was here.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/integration/curriculum_docs_test.rb [hunk 1]
```diff
@@ -131,6 +133,20 @@ def setup
       end
       assert_response :success
     end
+
+    test "class show should cache all queries" do
+      assert_cached_queries(0) do
```
Comment (maureensturgeon) [inline dashboard/test/integration/curriculum_docs_test.rb:138]: I don't understand the assertion here, the test is labeled "should cache all queries" but `assert_cached_queries(0)` makes it seem like there are 0 cached queries

**Review by maureensturgeon — APPROVED** [review]
Looks good, just one question!

File: dashboard/test/integration/curriculum_docs_test.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/test/integration/curriculum_docs_test.rb:138]: Oof yeah it is ambiguous. What it [actually does](https://github.com/code-dot-org/code-dot-org/blob/98f43456b6b72b1d140ec5a02335bcc40d5fec6d/dashboard/test/testing/capture_queries.rb#L32) is run the same block twice and assert the number of queries after the second time. So really, it should be something like `assert_queries_after_caching`.

Comment (maureensturgeon) [reply dashboard/test/integration/curriculum_docs_test.rb:138]: Ahh got it

**Pull request merged**